instrumentation/lettuce/lettuce-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_0/LettuceReactiveClientTest.java

```diff
@@ -0,0 +1,368 @@
+
+    Consumer<String> consumer =
+        res ->
+            testing.runWithSpan(
+                "callback",
+                () -> {
+                  assertThat(res).isEqualTo("OK");
+                  future.complete(res);
+                });
+
+    testing.runWithSpan(
+        "parent", () -> reactiveCommands.set("TESTSETKEY", "TESTSETVAL").subscribe(consumer));
+
+    await().untilAsserted(() -> assertThat(future).isCompletedWithValue("OK"));
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName("SET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                            equalTo(SemanticAttributes.DB_STATEMENT, "SET TESTSETKEY ?"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "SET")),
+                span ->
+                    span.hasName("callback")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0))));
+  }
+
+  @Test
+  void testGetCommandWithLambdaFunction() {
+    CompletableFuture<String> future = new CompletableFuture<>();
+
+    reactiveCommands
+        .get("TESTKEY")
+        .subscribe(
+            res -> {
+              assertThat(res).isEqualTo("TESTVAL");
+              future.complete(res);
+            });
+
+    await().untilAsserted(() -> assertThat(future).isCompletedWithValue("TESTVAL"));
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                            equalTo(SemanticAttributes.DB_STATEMENT, "GET TESTKEY"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "GET"))));
+  }
+
+  // to make sure instrumentation's chained completion stages won't interfere with user's, while
+  // still recording spans
+  @Test
+  void testGetNonExistentKeyCommand() {
+    CompletableFuture<String> future = new CompletableFuture<>();
+    String defaultVal = "NOT THIS VALUE";
+
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          reactiveCommands
+              .get("NON_EXISTENT_KEY")
+              .defaultIfEmpty(defaultVal)
+              .subscribe(
+                  res -> {
+                    testing.runWithSpan(
+                        "callback",
+                        () -> {
+                          assertThat(res).isEqualTo(defaultVal);
```

Comment:
I suspect that assertions might not work well in these callbacks, if the assertion fails the exception probably just gets swallowed. Could you check what happens if this assertion fails, if no useful info is provided you can just remove it.

Reply:
the tests fail when these assertions fail

![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/7630696/2583ad66-9486-43ba-be73-37513f7bbf5c)

Reply:
thanks for checking